### PR TITLE
Get Account by id improvement

### PR DESCRIPTION
**Pull request by JR-Morgan**
Code for finding an account by and throwing if not found was code that was duplicated in many places. 
I've created a new function in `AccountManager` for fetching an account by account id.

### CODE REVIEW

File: Core/Core/Credentials/Account.cs [hunk 1]
```diff
@@ -22,7 +25,7 @@ public string id
       {
         if (serverInfo == null || userInfo == null)
         {
-          throw new SpeckleException("Incomplete account info: cannot generate id.");
+          throw new InvalidOperationException("Incomplete account info: cannot generate id.");
```
Comment (AlanRynne) [inline Core/Core/Credentials/Account.cs:28]: Just wondering if we had any usages here were we try/catched SpeckleException, we should update those.

File: DUI3-DX/Connectors/Rhino/Speckle.Connectors.Rhino7/Operations/Send/RootObjectSender.cs [hunk 2]
```diff
@@ -37,10 +36,7 @@ public RootObjectSender(Func<Account, string, ITransport> transportFactory)
 
     onOperationProgressed?.Invoke("Uploading...", null);
 
-    // POC: FETCHING ACCOUNTS BY ID ONLY IS UNSAFE, we should filter by server first, but the server info is not stored on the ModelCard
```
Comment (AlanRynne) [inline DUI3-DX/Connectors/Rhino/Speckle.Connectors.Rhino7/Operations/Send/RootObjectSender.cs:40]: Ok I agree with this going out. I'll open a ticket for the lack of `serverUrl` being a limitation for document sharing between users.

File: DUI3-DX/Sdk/Speckle.Connectors.Utils/Operations/ReceiveOperation.cs [hunk 3]
```diff
@@ -32,26 +30,17 @@ public ReceiveOperation(IHostObjectBuilder hostObjectBuilder)
...
   {
     // 2 - Check account exist
-    Account account =
-      AccountManager.GetAccounts().FirstOrDefault(acc => acc.id == accountId)
-      ?? throw new SpeckleAccountManagerException();
+    Account account = AccountManager.GetAccount(accountId);
 
     // 3 - Get commit object from server
-    Client apiClient = new(account);
-    ServerTransport transport = new(account, projectId);
```
Comment (AlanRynne) [inline DUI3-DX/Sdk/Speckle.Connectors.Utils/Operations/ReceiveOperation.cs:41]: interesting that we never disposed of the transport. Could this have been the cause of some of our websocket issues?

**Review by AlanRynne — APPROVED** [review]

File: DUI3-DX/Sdk/Speckle.Connectors.Utils/Operations/ReceiveOperation.cs (hunk 3 above)
Comment (JR-Morgan) [reply DUI3-DX/Sdk/Speckle.Connectors.Utils/Operations/ReceiveOperation.cs:41]: Potentially, although, I'm not going to pretend I understand our web socket issue 🤣

File: Core/Core/Credentials/Account.cs (hunk 1 above)
Comment (JR-Morgan) [reply Core/Core/Credentials/Account.cs:28]: After a quick check, I couldn't find anywhere we are.

**Pull request merged**